### PR TITLE
make: reorganize makefile features

### PR DESCRIPTION
**Pull request by kaspar030**
This PR implements a chain ```Makefile.features``` taking any cpu, shared board and shared cpu file into account.

### CODE REVIEW

File: boards/arduino-atmega-common/Makefile.features [hunk 1]
```diff
@@ -4,6 +4,7 @@ FEATURES_PROVIDED += periph_i2c
 FEATURES_PROVIDED += periph_spi
 FEATURES_PROVIDED += periph_timer
 FEATURES_PROVIDED += periph_uart
+FEATURES_PROVIDED += periph_pm
```
Comment (haukepetersen) [inline boards/arduino-atmega-common/Makefile.features]: should be moved to the (possible shared) CPU features file (as implemented for all atmega CPUs, right?)

File: boards/f4vi1/Makefile.features [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # Put defined MCU peripherals here (in alphabetical order)
 FEATURES_PROVIDED += periph_timer
 FEATURES_PROVIDED += periph_uart
+FEATURES_PROVIDED += periph_gpio
```
Comment (haukepetersen) [inline boards/f4vi1/Makefile.features]: We could also move `periph_gpio` feature entries from all stm or even cortexm based boards to the CPU files, as these are not dependent on any board specific configuration and they are implemented for all these CPUs

File: cpu/cortexm_common/Makefile.features [hunk 3]
```diff
@@ -0,0 +1 @@
+FEATURES_PROVIDED += cpp
```
Comment (haukepetersen) [inline cpu/cortexm_common/Makefile.features]: Then this PR is missing the removal of all the `cpp` entries from all the cortexm boards, but this is part of the `WIP` right?

File: boards/arduino-atmega-common/Makefile.features (hunk 1 above)
Comment (kaspar030) [reply boards/arduino-atmega-common/Makefile.features]: done

File: boards/f4vi1/Makefile.features (hunk 2 above)
Comment (kaspar030) [reply boards/f4vi1/Makefile.features]: My understanding was that the feature being available also depends on the board's configuration in ```periph_conf.h``` and thus might override a CPU definition.

File: cpu/cortexm_common/Makefile.features (hunk 3 above)
Comment (kaspar030) [reply cpu/cortexm_common/Makefile.features]: correct!

Comment (kaspar030) [reply cpu/cortexm_common/Makefile.features]: done

**Comment by kaspar030** [comment]
Ready for review.

**Comment by kaspar030** [comment]
> It seems cpuid and hwrng are missing from kinetis_common

Yes, thanks. Fixed.

**Comment by haukepetersen** [comment]
Just wrote and used this test to check for changes in supported boards due to feature changes:
https://github.com/haukepetersen/riotsandbox/blob/master/tools/suptest/suptest.py
(Running it for master and for this branch, dumping the output in two files and looking at the diff of the two files)

Found that this PR introduces a changed output for 'make info-boards-supported' for a number of tests. See this gist for the diff: https://gist.github.com/haukepetersen/d295ff0bd76bd876a1ab38fd8930681c

**Comment by haukepetersen** [comment]
ups, the diff does not show the test names, so here is the raw data: https://gist.github.com/haukepetersen/fb6b50cff132e99790929976a6b4b8a7

**Comment by kaspar030** [comment]
I think I got them all. Actually, some more tests are build now (probably some features were not specified before.

btw, I used

```
#!/bin/sh

for dir in examples/* tests/*; do
    [ -d "$dir" ] && {
        for board in $(make --no-print-directory -C $dir info-boards-supported || echo ERROR); do
            echo $dir $board
        done
    }
done
```

on both master and this branch, and compared the output with

```
diff -U0 master.txt reorganize_makefile_features.txt
```

**Comment by haukepetersen** [comment]
notes for follow ups (do not need to be addressed in this PR):
- eliminate `FEATURES_MCU_GROUP` defines (or at least move them to CPU families)
- move `FEATURES_PROVIDED += periph_gpio` to CPU

**Comment by haukepetersen** [comment]
Oh: and you might want to sqash?! :-)

**Comment by kaspar030** [comment]
- squashed

@haukepetersen I kept the cpu reorganiation in seperate commits. That fine with you?

**Comment by haukepetersen** [comment]
> It seems cpuid and hwrng are missing from kinetis_common

This issue is addressed. So @gebart hope you are not mad that I dismiss your review :-)

**Comment by haukepetersen** [comment]
All good -> go

**Pull request merged**